src/components/Game/Game.jsx

```diff
@@ -0,0 +1,161 @@
+    }
+
+    fillSquare(i) {
+        const history = getHistory(this.state);
+        const currentSquares = getSquares(history);
+        if (!(calculateWinner(currentSquares) || currentSquares[i])) {
+            this.setState((prevState) => {
+                const prevHistory = getHistory(prevState);
+                const squares = [...getSquares(prevHistory)];
+                squares[i] = prevState.xIsNext ? GAME_SYMBOLS.X : GAME_SYMBOLS.O;
+                return {
+                    ...prevState,
+                    history: prevHistory.concat([{squares}]),
+                    xIsNext: !prevState.xIsNext,
+                    stepNumber: prevHistory.length,
+                }
+            });
+        }
+    }
+
+
+    handleClick(i) {
+        if (!(this.state.isMoveBlocked || this.state.isGameOver)) {
+            this.fillSquare(i);
+            const squares = this.state.history[this.state.stepNumber].squares;
+            const isMoveLast = squares.filter(el => el === null).length === 1;
+            if (this.state.botMode && !(isMoveLast || squares[i])) {
+                this.setState({isMoveBlocked: true});
+                setTimeout(() => {
+                    const botMoveIndex = calculateBotMove(this.state.history[this.state.history.length - 1].squares);
+                    this.setState({isMoveBlocked: false});
+                    this.fillSquare(botMoveIndex);
+                }, 500)
+            }
+        }
+    }
+
+    enableBotMode() {
+        this.setState({...initStartState(), botMode: true});
+    }
+
+    disableBotMode() {
+        this.setState({...initStartState(), botMode: false});
+    }
+
+    startNewGame() {
+        this.setState((prevState) => {
+            return {
+                ...initStartState(),
+                botMode: prevState.botMode,
+                score: prevState.score,
+            }
+        });
+    }
+
+    jumpTo(isNextMove) {
+        let step = isNextMove ? this.state.stepNumber + 1 : this.state.stepNumber - 1;
+        if (step % 2 && this.state.botMode && !this.state.isGameOver) {
+            isNextMove ? step += 1 : step -= 1;
+        }
+        if (this.state.history[step]) {
+            this.setState({
+                stepNumber: step,
+                xIsNext: !(step % 2),
+            })
+        }
+    }
+
+    resetScore() {
+        this.setState({
+            score: {
+                winsX: 0,
+                winsO: 0,
+            }
+        })
+    }
+
+    render() {
+        const {
+            history,
+            stepNumber,
+            xIsNext,
+            isGameOver
+        } = this.state;
+        const currentSquares = history[stepNumber].squares;
+        const winner = calculateWinner(currentSquares);
+        const status = winner ? `Winner is ${winner}`
+            : ((history.length - 1 === stepNumber && isGameOver) ? 'Draw' : `Next move: ${xIsNext ? GAME_SYMBOLS.X : GAME_SYMBOLS.O}`);
```

Comment:
create methods for these calc-s, try to restructure code to not use nested ternary operator

Reply:
```suggestion
render() {
        const {history, stepNumber} = this.state;
        const currentSquares = getSquares(history, stepNumber);
        const winner = calculateWinner(currentSquares);
        const status = getStatus(winner, this.state);
        ...
```
Now in render() I am only calling methods and returning jsx
The code of getStatus method:
```suggestion
export function getStatus(winner, state) {
    const {
        history,
        stepNumber,
        xIsNext,
        isGameOver
    } = state;
    if (winner) {
        return `Winner is ${winner}`;
    } else {
        if (history.length - 1 === stepNumber && isGameOver) {
            return 'Draw';
        } else {
            return `Next move: ${xIsNext ? GAME_SYMBOLS.X : GAME_SYMBOLS.O}`
        }
    }
}
```